### ISSUE TITLE
Build fails with GraphQL enum name started with underscore and digit

### ISSUE THREAD

**Issue by spuliaiev** [apollographql/apollo-ios#1088]
iOS build fails if a GraphQL enumerator type field has a value with name started with an underscore and digit; error message on build

`'K' is not a valid digit in integer literal`

Generated code:

```
public enum ColorEffect: RawRepresentable, Equatable, Hashable, CaseIterable, Apollo.JSONDecodable, Apollo.JSONEncodable {
  public typealias RawValue = String
  case metallic
  case matte
  case glossy
  case flipFlop
  case nacre
  case pearl
  case candy
  case lotus
  case foiled
  case uni
  case 2KMetallic
  /// Auto generated constant for unknown enum values
  case __unknown(RawValue)
```

schema:

```
enum ColorEffect {
  METALLIC
  MATTE
  GLOSSY
  FLIP_FLOP
  NACRE
  PEARL
  CANDY
  LOTUS
  FOILED
  UNI
  _2K_METALLIC
}
```

Could you help with this?

**Comment by designatednerd** [comment]
Oof. Yeah the underscore is getting stripped by our typescript-based codegen, and you can't have enum cases that start with a digit even if you backtick-escape them. 

This is something that's been addressed in the experimental swift codegen, but enums are the only thing that I've actually got generating so far, so that's not actually going to be all that helpful at the moment.

A real solution is going to be adding some kind of option to not strip underscores off enum cases in the typescript codegen.

**Comment by calvincestari** [comment]
Hi 👋🏻 - this issue should be resolved in 1.0. 1.0 is currently in alpha releases with a full 1.0 release in H2 2022. Please take a look at the [latest alpha](https://github.com/apollographql/apollo-ios/releases) and let us know if this is still an issue for you.
